client/src/routes/Session/components/Participants/Participants.tsx

```diff
@@ -31,6 +55,32 @@ const Participants: React.FC<ParticipantsProps> = ({participants}) => {
     ),
   );
 
+  useEffect(() => {
+    if (participants.length < 2) {
+      return;
+    }
+    if (participants.length > numberOfParticipants) {
+      setNumberOfParticipants(participants.length);
+      setJoinMessage(
+        t('participant.joined', {
+          user: participants[participants.length - 1].user_name,
+        }),
+      );
+      setTimeout(() => {
+        setJoinMessage('');
+      }, 4000);
+      return clearTimeout(joinMessage);
+    }
+    if (participants.length < numberOfParticipants) {
+      setNumberOfParticipants(participants.length);
+      setJoinMessage(t('participant.left'));
+      setTimeout(() => {
+        setJoinMessage('');
+      }, 4000);
+      return clearTimeout(joinMessage);
```

Comment:
First of all, this will be a nice UX improvement - thank you! 💚 

Just noticed here that the `clearTimeout()` is trying to clear `joinMessage` which is the string state, rather than references to the active timeouts.

What you probably meant to do is to keep the timeout ids returned by `setTimeout()` on line 77 and line 69 and pass those to `clearTimeout()` instead, right?

Also the cleanup function of the useEffect at line 80 should probably be a callback function, like `() => { clearTimeout() }`. Otherwise it would try to use the result of calling `clearTimeout()` right away when the effect is first created.